src/lib/useSyncPoints.js

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+import { isEqual } from 'lodash';
+
+import { usePointCache } from 'store/pointCache';
+import usePreviousValue from 'lib/usePreviousValue';
+
+// sync a known point to the cache (i.e display values)
+export function useSyncKnownPoints(points = []) {
+  //
+}
+
+// sync an owned point to the cache
+export function useSyncOwnedPoints(points = []) {
+  const { syncOwnedPoint } = usePointCache();
+  const prevPoints = usePreviousValue(points);
+
+  useEffect(() => {
+    if (!isEqual(prevPoints, points)) {
+      // if the actual set of points changes, sync the whole array
+      // TODO: we should get clever about caching
+      points.forEach(point => syncOwnedPoint(point));
```

Comment:
Yeah this _looks_ like it re-builds the entire cache every time a new point is added? But maybe I misunderstand.

Reply:
correct — to be fair, the old code was doing the same thing every time the points page was mounted. this code should only update the entire list of points when a new point is added. the clever code can be part of the actual `sync*` functions that decide whether or not to fetch from the chain